File: examples/client/mq-python-wrapper.js
var SgMessagingClient = require('sg-messaging-client');
var sgMessagingClient = new SgMessagingClient();

var zerorpc = require('zerorpc');

var client = new zerorpc.Client({
	timeout: 1 * 60 * 5
});
client.connect('tcp://127.0.0.1:4242');

client.invoke('COMMAND_LIST', {}, function (e, res, streaming) {
	if(e) {
		return console.log(e);
	}

	function commandWrapper (commands, command) {
		return function (input, progressCallback, callback) {
			client.invoke(command, input, callback);
		};
	}

	var commands = {};
	var parameters;
	for(var command in res) {
		parameters = res[command];

		sgMessagingClient().subscribe(command, commandWrapper(commands, command), parameters);
	}

	var server = new zerorpc.Server(commands);
	server.bind('tcp://127.0.0.1:4241');

	server.on('error', function (e) {
		console.log(e);
	});

});

File: examples/client/mq-python-wrapper.py
import zerorpc
import logging
from IPython.core.debugger import Tracer
debug_here = Tracer()
logging.basicConfig()

client = zerorpc.Client(timeout=(1 * 60 * 5))
client.connect('tcp://127.0.0.1:4241')

def provideCommandList (commandNames):

	commands = {}
	commandName = None

	for i in range(len(commandNames)):
		commandName = commandNames[i]
		commands[commandName] = {
			'progress': True
		}

	def responseCommandList (a):
		print '\n> responseCommandList():'
		print commands

		return commands

	return responseCommandList

def main (commands):
	commandNames = commands.keys()
	commands['COMMAND_LIST'] = provideCommandList(commandNames)

	print '\n> main():'
	print commands

	s = zerorpc.Server(commands)
	s.bind('tcp://127.0.0.1:4242')
	s.run()

def TESTY (data):
	print '\n> testy()'
	print data

	num = data['num']
	return 'hello, ' + str(num)

main({
	'hello': TESTY
})
